File: server/src/main/java/moe/yuuta/server/api/ApiHandler.java
package moe.yuuta.server.api;

import io.vertx.core.Vertx;
import io.vertx.ext.web.RoutingContext;
import moe.yuuta.server.github.GitHubApi;
import moe.yuuta.server.mipush.MiPushApi;

public interface ApiHandler {
    static ApiHandler apiHandler(Vertx vertx) {
        return new ApiHandlerImpl(vertx);
    }

    void handlePush (RoutingContext routingContext);
    void handleFrameworkIndex(RoutingContext routingContext);
    void handleTesterIndex (RoutingContext routingContext);
    MiPushApi getMiPushApi ();
    void handleUpdate(RoutingContext routingContext);
    GitHubApi getGitHubApi ();
}

File: server/src/main/kotlin/moe/yuuta/server/api/ApiHandler.kt
package moe.yuuta.server.api

import io.vertx.core.Vertx
import io.vertx.ext.web.RoutingContext

interface ApiHandler {
    companion object {
        @JvmStatic
        fun apiHandler(vertx: Vertx?): ApiHandler {
            return ApiHandlerImpl(vertx)
        }
    }

    fun handlePush (routingContext: RoutingContext)
    fun handleFrameworkIndex(routingContext: RoutingContext)
    fun handleTesterIndex(routingContext: RoutingContext)
    fun handleUpdate(routingContext: RoutingContext)
    fun handleGetTopicList(routingContext: RoutingContext)
}
